### PR TITLE
Add validation rule `CommandSenderMustBeAnExistingMarketParticipant`

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

The validation rule `CommandSenderMustBeAnExistingMarketParticipant` is executed for all charge types and operation types. It will cause charge commands to be rejected and thus rejections to be sent to the message sender through post office.

## References

* #117

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/Application/ChangeOfCharges/ChangeOfChargePipelineTests.cs [hunk 1]
```diff
@@ -95,7 +95,7 @@ public ChangeOfChargePipelineTests([NotNull] DbContextRegistrator dbContextRegis
         }
 
         [Theory(Timeout = 120000)]
-        [InlineAutoMoqData("TestFiles/InvalidTariffAddition.json")]
+        [InlineAutoMoqData("TestFiles/InValidCreateTariffCommand.json")]
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/Application/ChangeOfCharges/ChangeOfChargePipelineTests.cs:98]: I think the capital V in InValid should be removed to get the right file

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/BusinessValidation/Factories/BusinessUpdateValidationRulesFactory.cs [hunk 2]
```diff
@@ -48,27 +52,34 @@ public async Task<IValidationRuleSet> CreateRulesForUpdateCommandAsync([NotNull]
             var commandChargeTypeOwnerMRid = chargeCommand.ChargeOperation.ChargeOwner;
 
             var charge = await _chargeRepository.GetChargeAsync(chargeTypeMRid, commandChargeTypeOwnerMRid).ConfigureAwait(false);
-
             if (charge == null)
             {
-                throw new Exception($"Charge found on MRid: {chargeTypeMRid}, ChargeTypeOwnerMRid: {commandChargeTypeOwnerMRid}");
+                throw new Exception($"Charge not found on MRid: {chargeTypeMRid}, ChargeTypeOwnerMRid: {commandChargeTypeOwnerMRid}");
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/BusinessValidation/Factories/BusinessUpdateValidationRulesFactory.cs:57]: Thx. I'm however not going to do the other things as they are completely unrelated to this PR and issue and would require adding unit testing and repeating manual testing.

**Review by prtandrup — COMMENTED** [review]
Minor things

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/BusinessValidation/Factories/BusinessCreateValidationRulesFactoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    public class BusinessCreateValidationRulesFactoryTests
+    {
+        [Theory]
+        [InlineAutoMoqData(typeof(CommandSenderMustBeAnExistingPartyRule), ChargeType.Subscription)]
+        [InlineAutoMoqData(typeof(CommandSenderMustBeAnExistingPartyRule), ChargeType.Fee)]
+        [InlineAutoMoqData(typeof(CommandSenderMustBeAnExistingPartyRule), ChargeType.Tariff)]
+        public async Task CreateRulesForCreateCommandAsync_ReturnsRulesContainingExpectedRuleType(
+            Type expectedRuleType,
+            ChargeType chargeType,
+            [NotNull] [Frozen] Mock<IRulesConfigurationRepository> updateRulesConfigurationRepository,
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/BusinessValidation/Factories/BusinessCreateValidationRulesFactoryTests.cs:43]: Was this supposed to be...?

```suggestion
            [NotNull] [Frozen] Mock<IRulesConfigurationRepository> createRulesConfigurationRepository,
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/Application/ChangeOfCharges/ChangeOfChargePipelineTests.cs (hunk 1 above)
Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/Application/ChangeOfCharges/ChangeOfChargePipelineTests.cs:98]: The capital V error is already fixed on main branch

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs [hunk 4]
```diff
@@ -42,5 +42,6 @@ public enum ValidationRuleIdentifier
         ChargePriceMaximumDigitsAndDecimals, // VR457
         FeeMustHaveSinglePrice, // VR507-2
         SubscriptionMustHaveSinglePrice, // VR507-2
+        CommandSenderMustBeAnExistingParty, // VR152
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs:45]: Even though it is longer val name, I would prefer the rule to be `CommandSenderMustBeAnExistingMarketParticipant` to be as consistent as possible.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/Application/ChangeOfCharges/ChangeOfChargeLocalHostTests.cs [hunk 5]
```diff
@@ -133,7 +133,7 @@ public ChangeOfChargeLocalHostTests(ITestOutputHelper testOutputHelper, [NotNull
         }
 
         [Theory(Skip = "Run at localhost to integration test local code using sandboxed Service Bus")]
-        [InlineAutoMoqData("TestFiles/InvalidTariffAddition.json")]
+        [InlineAutoMoqData("TestFiles/InValidCreateTariffCommand.json")]
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/Application/ChangeOfCharges/ChangeOfChargeLocalHostTests.cs:136]: InValid -> Invalid

**Comment by BjarkeMeier** [comment]
> Minor things

Thank you for feedback @prtandrup. I haven't committed any of your suggestions directly because they would break the build. But I have fixed all of them.

**Review by prtandrup — APPROVED** [review]
LGTM

**Review by HenrikSommer — APPROVED** [review]

**Pull request merged**